src/nuis/measurement/Projection.h

```diff
@@ -0,0 +1,195 @@
+// Copyright 2016-2021 L. Pickering, P Stowell, R. Terri, C. Wilkinson, C. Wret
+
+/*******************************************************************************
+ *    This file is part of NUISANCE.
+ *
+ *    NUISANCE is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    NUISANCE is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with NUISANCE.  If not, see <http://www.gnu.org/licenses/>.
+ *******************************************************************************/
+#pragma once
+#include <vector>
+#include <string>
+#include <iostream>
+#include <memory>
+#include <map>
+
+#include <Eigen/Dense>
+
+#include "yaml-cpp/yaml.h"
+using namespace YAML;
+
+#include "nuis/measurement/Variables.h"
+#include "nuis/measurement/Document.h"
+
+enum likelihood_form_enum {
+     kFullChi2,
+     kShapeChi2,
+     kNormChi2,
+     kShapePlusNormChi2,
+     kEvents
+};
+
+namespace nuis {
+namespace measurement {
+
+enum bands_enum {
+     kDataBand = 0,
+     kMCBand
+};
+
+enum band_selection_enum {
+     kAllMC,
+     kAll,
+     kExceptData
+};
+
+struct Band {
+
+     Band(){}
+
+     explicit Band(int nbins,
+          std::string lab = "", int bid = 1) {
+          band_id = bid;
+          label = lab;
+          value = Eigen::ArrayXd(nbins);
+          count = Eigen::ArrayXi(nbins);
+          error = Eigen::ArrayXd(nbins);
+     }
+
+     std::string label;
+     int band_id;
+     Eigen::ArrayXd value;
+     Eigen::ArrayXi count;
+     Eigen::ArrayXd error;
+};
+
+
+class Projection {
+  public:
+     Projection();
+
+     explicit Projection(YAML::Node config);
+
+     ~Projection();
+
+     void Fill(int signal,
+          const std::vector<double>& indep_vals, double w = 1.0);
+
+     void Scale(double v, int band = kMCBand);
+
+     void Reset(int band = kExceptData);
+
+     // New bin structure scheme
+     std::vector<std::string> axis_label;
```

Comment:
is there substructure here that can be expressed via a local type:

```c++
struct labels {
  std::string label,sublabel;
};
std::vector<labels> .. 
```


or:
```c++
struct labels {
  std::string label;
  std::vector<sublabel>;
};
std::vector<labels> .. 
```

or such? If there are lists of 'index-synched' things it feels neater to keep them together unless there is a good reason not to.